Add -language:xx argument

xenicle opened the pull request:
Add -language:xx argument to changes the game's language. example: -language:en,
supported languages: de,en,es,fr,it,ja,ko,pl,pt,ru,tc,uk,zh.

xenicle committed:
Add -language:xx argument

Add -language:xx argument to changes the game's language. example: -language:en,
supported languages: de,en,es,fr,it,ja,ko,pl,pt,ru,tc,uk,zh.

ananace commented on WF_Proton_Launcher/Main.cpp:
```diff
@@ -105,6 +105,12 @@ std::string registry_option;
+
+std::string lang;
+
+std::string supported_languages[13] = {"de", "es", "fr", "it", "ja", "ko", "pl", "pt", "ru", "tc", "uk", "zh", "en"};
```
If this is required then I'd prefer if this was `const` and all uppercase, preferrably also relocated up top to the rest of the internal constants.

ananace commented on WF_Proton_Launcher/Main.cpp:
```diff
@@ -130,6 +136,13 @@ int main(int argc, char** argv)
+			lang = arg.substr(10, 2);
+			auto found = std::find(std::begin(supported_languages), std::end(supported_languages), lang);
+			if (found != std::end(supported_languages))
+				language = arg;
```
To repeat the complaint I had on the similar change for supporting selecting the registry;

With the code as it is here, if Warframe was to add localization for another language - Swedish for instance, then the launcher code would have to be updated to add `"se"` to the list of valid languages. If we just pass along any arbitrary code that the user specifies then we automatically become forward-compatible.  
So if the validation was skipped here, what would warframe do if it receives a launch argument to run with - for example - `-language:wf`? Would it give an error or would it silently ignore the invalid argument?

ananace replied:
In general, I'd much prefer if we leave validation up to the ones who control the exact list of valid inputs, that way we won't silently ignore valid input - or worse, give false positive errors on said valid input.

ananace commented on WF_Proton_Launcher/Main.cpp:
```diff
@@ -105,6 +105,12 @@ std::string registry_option;
 
 std::string fullscreen_option = " -fullscreen:0 ";
 
+std::string language = "-language:en";
```
No ending space, will cause issues in the string concatenation.

xenicle committed:
Change to add -language:xx without validation

Change to add -language:xx without validation we leave validation up to the ones who control the exact list of valid inputs, that way we won't silently ignore valid input - or worse, give false positive errors on said valid input.

ananace commented on WF_Proton_Launcher/Main.cpp:
```diff
@@ -142,6 +146,8 @@ int main(int argc, char** argv)
 				<< "  -v  Print more verbose output" << std::endl
 				<< "  -32 Launch the 32-bit binary instead of the 64-bit one" << std::endl
 				<< "  -fullscreen Launch the game in full-screen" << std::endl
+				<< "  -language:xx changes the game's language. example: -language:en" << std::endl
```
Nit; I'd change "example:" to instead say "Default:", that clarifies both as an example and as the default value.

xenicle replied:
Alright !

xenicle committed:
Change example to Default in -language:xx usage.
